kythe/cxx/indexer/cxx/GraphObserver.h

```diff
@@ -1119,10 +1149,9 @@ class GraphObserver {
   clang::LangOptions* LangOptions = nullptr;
   clang::Preprocessor* Preprocessor = nullptr;
   ProfilingCallback ReportProfileEvent = [](const char*, ProfilingEvent) {};
+  std::unique_ptr<HashRecorder> hash_recorder_;
```

Comment:
This seems more naturally and unowned pointer if it is always being passed in externally.